### PR TITLE
Release EUIs of deleted gateways

### PR DESCRIPTION
**Pull request by bafonins**
#### Summary

Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/1120

#### Changes

- Add `afterDelete` hook for gateways to reset their eui.
- Update gateway deletion modal text.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/views/gateway-general-settings/index.js [hunk 1]
```diff
@@ -42,8 +42,10 @@ import { mayEditBasicGatewayInformation, mayDeleteGateway } from '../../lib/feat
 const m = defineMessages({
   updateSuccess: 'Successfully updated gateway',
   deleteGateway: 'Delete Gateway',
-  modalWarning:
-    'Are you sure you want to delete "{gtwName}"? This action cannot be undone and it will not be possible to reuse the gateway ID and EUI!',
+  modalWarning: `
+Are you sure you want to delete "{gtwName}"? This action cannot be undone and it will not be possible to reuse the gateway ID!
+Note, deleting this gateway will release its EUI and will make it available for registering a new one with the same gateway EUI.
```
Comment (kschiffer) [inline pkg/webui/console/views/gateway-general-settings/index.js]: Do we need to be this explicit?

Comment (htdvisser) [reply pkg/webui/console/views/gateway-general-settings/index.js]: I'd say no. Let's make it consistent with the message for other entities.

**Comment by t0mZ4** [comment]
Hi

I still see an issue when I use the CLI to delete a gateway, and then I cannot re-add the gateway with the same id.

```

error:pkg/identityserver/store:id_taken (ID already taken)
    correlation_id=fc206b725e0543d691dafa5a4d8d2674

```

any way to workaround this!?

**Comment by htdvisser** [comment]
The EUI was released when you deleted the gateway, but the ID is still reserved.

This is for security reasons, as historical data (in external integrations) is linked to the ID, and if someone else would register a gateway with the same ID after you delete it, they would have access to your historical data, which probably isn't what you want.

With #3467 it's possible for admin users to release the IDs, but of course that could cause some security issues like the one I described above.

**Pull request merged**